### ISSUE TITLE
DateMatcher and MultiDateMatcher getAnchorDateMonth returns incorrect value after setting the parameter

### ISSUE THREAD

**Issue by dkincaid** [JohnSnowLabs/spark-nlp#5673]
## Description
After creating a DateMatcher or MultiDateMatcher annotator and using `setAnchorDateMonth()` to set the parameter, a call to `getAnchorDateMonth()` returns a different value than was set. It appears to return one less than whatever integer was set.

```java
DateMatcher myMatcher = new DateMatcher();
myMatcher.setAnchorDateMonth(5);
int wrongMonth = myMatcher.getAnchorDateMonth();
// wrongMonth will be 4 instead of 5
```

## Context
Adding these parameters to the DateMatcher annotator implementation in the R package sparknlp

## Your Environment

* Spark NLP version `sparknlp.version()`: 3.0.3
* Apache NLP version `spark.version`: 3.1.1
* Java version `java -version`: OpenJDK 1.8.0_292
* Operating System and version: Linux Pop_OS 20.10
* Link to your project (if any): https://github.com/r-spark/sparknlp

**Comment by maziyarpanahi** [comment]
Thanks @dkincaid for reporting this. I thought this was fixed in 3.0.3 https://github.com/JohnSnowLabs/spark-nlp/pull/2848

@wolliq I know you are very familiar with DateMatcher and MultiDateMatcher. Could you please test this in all 3 languages of Python, Scala, and Java?

**Comment by wolliq** [comment]
Hi @dkincaid,

code fix has been pushed for the next release integration.
Once integrated, could you please close the issue?

Thank you
